### ISSUE TITLE
Add support for other backends

### ISSUE THREAD

**Issue by crimsonknave** [jmpsec/osctrl#86]
According to https://osctrl.net/components/backend/ it wouldn't be much work to support backends other than postgres. I'd love to see mysql support as that's what my database admins support.

**Comment by javuto** [comment]
Just removed the requirement for jsonb so any backend could be used now, via gorm

**Comment by javuto** [comment]
PR for the refactoring: https://github.com/jmpsec/osctrl/pull/228

**Comment by javuto** [comment]
Dependencies for PostgreSQL have been removed, it should be straight forward to add support to other backends. Is there still any interest for it? Thanks!

**Comment by crimsonknave** [comment]
I'm no longer on a team that deals with osquery, so I don't have a need for it, but it does seem useful in general to me.

**Comment by stfast** [comment]
> Dependencies for PostgreSQL have been removed, it should be straight forward to add support to other backends. Is there still any interest for it? Thanks!

Yes, there is an interest, as from system administration point of view with MySQL is way-way much easier to maintain HA system with master-master replication, which is a nightmare in PostgreSQL.  Also PostgreSQL upgrades are a nightmare, since you can not skip major versions with distributions (like Fedora 33 directly to Fedora 41, which you can do, but not with PostgreSQL service).

It would be great to put mysql or postgres as a DSN parameter.

**Comment by javuto** [comment]
Implemented in #708
